hail/src/main/scala/is/hail/backend/service/Worker.scala

```diff
@@ -160,14 +160,14 @@ object Worker {
     }
 
     var result: Array[Byte] = null
-    var userError: HailException = null
+    var errorWhileExecutingUserCode: Exception = null
     using(new ServiceTaskContext(i)) { htc =>
       try {
         retryTransientErrors {
           result = f(context, htc, theHailClassLoader, fs)
         }
       } catch {
-        case err: HailException => userError = err
+        case err: Exception => errorWhileExecutingUserCode = err
```

Comment:
I don't think you should be catching all exceptions. Perhaps use `case NonFatail(err)`.

Reply:
Hmm. I'm catching `Exception`, not `Throwable`. [Of the exceptions that `NonFatal` ignores](https://github.com/scala/scala/blob/v2.12.18/src/library/scala/util/control/NonFatal.scala#L41), only `InterruptedException` is an `Exception`. I think you're right about interrupted exception though; if a JVMJob is cancelled we cancel the worker thread which probably interrupts it. I'll change to NonFatal.